File: frontend/src/types/complaint.ts
export interface Session {
  session_id: string;
  created_at: string;
  last_updated: string;
  status: string;
}

export interface Complaint {
  company_name: string | null;
  manufacturer: string | null;

  product_name: string | null;
  generic_name: string | null;
  strength: string | null;
  dosage_form: string | null;
  pack_size: string | null;

  batch_number: string | null;
  manufacturing_date: string | null;
  expiry_date: string | null;

  quantity: string | null;

  complaint_description: string | null;
  complaint_category: string | null;
  complaint_type: string | null;
  defect_type: string | null;

  severity: string | null;

  reported_event: string | null;
  symptoms: string[];
}

export interface AI {
  summary: string | null;
  confidence: number | null;

  missing_fields: string[];

  next_question?: string | null;

  extraction_status?: string | null;

  reasoning?: string | null;
}

export interface ValidationError {
  field: string;
  message: string;
}

export interface Validation {
  is_valid: boolean;
  errors: ValidationError[];
  warnings: string[];
  missing_fields: string[];
  validation_timestamp: string | null;
}

export interface Risk {
  priority: string | null;
  score: number | null;
  risk_factors: string[];
  reasons: string[];
  recommended_actions: string[];
  assessment_timestamp: string | null;
}

export interface ComplaintState {
  session: Session;
  complaint: Complaint;
  ai: AI;
  validation: Validation;
  risk: Risk;
}

export interface ComplaintResponse {
  success: boolean;
  message: string;
  data: ComplaintState;
}

export interface ComplaintCreateRequest {
  message: string;
}

File: backend/app/models/complaint.py
"""Complaint sub-model of ComplaintState."""

from datetime import date

from pydantic import BaseModel, Field


class Complaint(BaseModel):
    """Structured complaint information captured during intake."""

    # ==========================
    # Company Information
    # ==========================
    company_name: str | None = None
    manufacturer: str | None = None

    # ==========================
    # Product Information
    # ==========================
    product_name: str | None = None
    generic_name: str | None = None
    strength: str | None = None
    dosage_form: str | None = None
    pack_size: str | None = None
    quantity: str | None = None

    # ==========================
    # Batch Information
    # ==========================
    batch_number: str | None = None
    manufacturing_date: date | None = None
    expiry_date: date | None = None

    # ==========================
    # Complaint Information
    # ==========================
    complaint_description: str | None = None
    complaint_category: str | None = None
    complaint_type: str | None = None
    defect_type: str | None = None
    severity: str | None = None

    # ==========================
    # Patient Information
    # ==========================
    reported_event: str | None = None
    symptoms: list[str] = Field(default_factory=list)